omod/src/main/java/org/openmrs/module/idgen/rest/resource/AutoGenerationOptionResource.java

```diff
@@ -0,0 +1,232 @@
+	@PropertyGetter("uuid")
+	public String getUuid(AutoGenerationOption autoGenerationOption) {
+		return autoGenerationOption.getUuid();
+	}
+	
+	@PropertyGetter("display")
+	public String getDisplayString(AutoGenerationOption autoGenerationOption) {
+		return autoGenerationOption.getIdentifierType() + " - " + autoGenerationOption.getSource() + " - "
+		        + autoGenerationOption.getLocation();
+	}
+	
+	@Override
+	public DelegatingResourceDescription getCreatableProperties() {
+		DelegatingResourceDescription description = new DelegatingResourceDescription();
+		description.addProperty("uuid");
+		description.addProperty("identifierType");
+		description.addProperty("location");
+		description.addProperty("source");
+		description.addProperty("manualEntryEnabled");
+		description.addProperty("automaticGenerationEnabled");
+		return description;
+	}
+	
+	@Override
+	public DelegatingResourceDescription getUpdatableProperties() {
+		DelegatingResourceDescription description = new DelegatingResourceDescription();
+		description.addProperty("location");
+		description.addProperty("source");
+		description.addProperty("manualEntryEnabled");
+		description.addProperty("automaticGenerationEnabled");
+		return description;
+	}
+	
+	@Override
+	public SimpleObject create(SimpleObject postBody, RequestContext context) throws ResponseException {
+		try {
+			String manualEntryEnabled = postBody.get("manualEntryEnabled").toString();
+			String automaticGenerationEnabled = postBody.get("automaticGenerationEnabled").toString();
+			String sourceUuid = postBody.get("source").toString();
+			String identifierUuid = postBody.get("identifierType").toString();
+			String locationUuid = ((postBody.get("location") != null) && (!postBody.get("location").equals("")))
+			        ? postBody.get("location").toString()
+			        : null;
+			AutoGenerationOption autoGenerationOption = new AutoGenerationOption();
+			IdentifierSource identifierSource = identifierSourceService.getIdentifierSourceByUuid(sourceUuid);
+			PatientIdentifierType patientIdentifierType = Context.getPatientService()
+			        .getPatientIdentifierTypeByUuid(identifierUuid);
+			autoGenerationOption.setSource(identifierSource);
+			autoGenerationOption.setIdentifierType(patientIdentifierType);
+			autoGenerationOption.setManualEntryEnabled(Boolean.parseBoolean(manualEntryEnabled));
+			autoGenerationOption.setAutomaticGenerationEnabled(Boolean.parseBoolean(automaticGenerationEnabled));
+			if (locationUuid != null) {
+				Location location = Context.getLocationService().getLocationByUuid(locationUuid);
+				autoGenerationOption.setLocation(location);
+				return (SimpleObject) ConversionUtil.convertToRepresentation(save(autoGenerationOption),
+				    Representation.FULL);
+			}
+			return (SimpleObject) ConversionUtil.convertToRepresentation(save(autoGenerationOption), Representation.DEFAULT);
+		}
+		catch (NullPointerException e) {
+			throw new ResourceDoesNotSupportOperationException(
```

Comment:
Where does this null pointer exception come from?

Reply:
it arises when the property getters are given a null argument eg in line 137 if the sourceUuid is null the getIdentifierSourceByUuid() will throw a null pointer exception

Reply:
Why do you make the getIdentifierSourceByUuid call if you have a null sourceUuid? Would you expect it to magically return something?